### PR TITLE
Coverity Scan issues CID-65554, 481888, 48193

### PR DESCRIPTION
**Pull request by hlaccabu**
Fixing small issues found by coverity scan, all located in src/shim/
Uncaught exception in hwctx.cpp, parse warning in bo.cpp, AUTO_CAUSES_COPY in kmq/bo.cpp

### CODE REVIEW

File: src/shim/bo.cpp [hunk 1]
```diff
@@ -8,7 +8,7 @@
 namespace {
 
 uint32_t
-alloc_drm_bo(const shim_xdna::pdev& dev, amdxdna_bo_type type, void* buf, size_t size)
+alloc_drm_bo(const shim_xdna::pdev& dev, uint32_t type, void* buf, size_t size)
```
Comment (xdavidz) [inline src/shim/bo.cpp]: I'd keep the amdxdna_bo_type here for this api even it is prviate anonymous namespace.

And cast it to u32 at line16. Does that make sense?

Comment (hlaccabu) [reply src/shim/bo.cpp]: I tried it, the compiler doesn't like it it won't build that way.

**Review by keryell — CHANGES_REQUESTED** [review]
It would be great to have commit messages that make sense for an open-source project.

File: src/shim/bo.cpp (hunk 1 above)
Comment (xdavidz) [reply src/shim/bo.cpp]: > I tried it, the compiler doesn't like it it won't build that way.

this should work:
.type = static_cast<uint32_t>(type),

**Review by keryell — APPROVED** [review]
Thanks!

**Review by xdavidz — APPROVED** [review]

**Pull request merged**